### PR TITLE
(React) movies-list-filter

### PR DESCRIPTION
**Pull request by Ostafii4uk**
[DEMO LINK](https://Ostafii4uk.github.io/react_movies-list-filter/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,9 +1,20 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+
+  const visibleMovies = [...moviesFromServer].filter(movie => {
```
Comment (AnnSerdechna) [inline src/App.tsx:9]: [...moviesFromServer] - no need yo make a copy, filter return new array

**Review by AnnSerdechna — APPROVED** [review]
Good work 👌

File: src/App.tsx [hunk 2]
```diff
@@ -1,9 +1,20 @@
...
+    const movieTitleLowerCase = movie.title.toLowerCase();
+    const movieDescriptionLowerCase = movie.description.toLowerCase();
+    const queryLowerCase = query.toLowerCase();
+
+    return movieTitleLowerCase.includes(queryLowerCase)
+     || movieDescriptionLowerCase.includes(queryLowerCase);
+  });
```
Comment (Ihor-Karpyn) [inline src/App.tsx:16]: Agree, you shouldn't make a copy. Also, create function to filter movies and use here `useMemo`

**Review by Ihor-Karpyn — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 1 above)
Comment (Ostafii4uk) [reply src/App.tsx:9]: 👍

File: src/App.tsx (hunk 2 above)
Comment (Ostafii4uk) [reply src/App.tsx:16]: Ok

File: src/App.tsx [hunk 3]
```diff
@@ -1,9 +1,23 @@
-import React from 'react';
+import React, { useMemo, useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+
+  const filteredMovies = (movies: Movie[]) => (
```
Comment (Ihor-Karpyn) [inline src/App.tsx:9]:
```suggestion
  const filterMovies = (movies: Movie[]) => (
```

File: src/App.tsx [hunk 4]
```diff
@@ -1,9 +1,23 @@
...
+    movies.filter(movie => {
+      const movieTitleLowerCase = movie.title.toLowerCase();
+      const movieDescriptionLowerCase = movie.description.toLowerCase();
+      const queryLowerCase = query.toLowerCase();
+
+      return movieTitleLowerCase.includes(queryLowerCase)
+      || movieDescriptionLowerCase.includes(queryLowerCase);
+    }));
+
+  const visibleMovies = useMemo(() => filteredMovies(moviesFromServer), [query]);
```
Comment (Ihor-Karpyn) [inline src/App.tsx:19]:
```suggestion
  const filteredMovies = useMemo(() => filteredMovies(moviesFromServer), [query]);
```

**Review by Ihor-Karpyn — APPROVED** [review]

**Pull request closed without merging**